File: src/main/java/de/byjoker/myjfql/command/ShutdownCommand.java
package de.byjoker.myjfql.command;

import de.byjoker.myjfql.core.MyJFQL;

import java.util.Collections;
import java.util.List;
import java.util.Map;

@CommandHandler
public class ShutdownCommand extends ConsoleCommand {

    public ShutdownCommand() {
        super("shutdown", Collections.singletonList("COMMAND"));
    }

    @Override
    public void executeAsConsole(ConsoleCommandSender sender, Map<String, List<String>> args) {
        MyJFQL.getInstance().shutdown();
    }

}

File: src/main/java/de/byjoker/myjfql/command/ShutdownCommand.kt
package de.byjoker.myjfql.command

import de.byjoker.myjfql.core.MyJFQL
import de.byjoker.myjfql.user.Permission
import de.byjoker.myjfql.user.UserType
import org.jline.reader.ParsedLine
import kotlin.system.exitProcess

@CommandHandler
class ShutdownCommand : Command("shutdown", listOf("command", "hardly"), listOf("stop", "exit")) {

    override fun execute(sender: CommandSender, args: Map<String, List<String>>) {
        if (!sender.permitted(Permission.SHUTDOWN)) {
            sender.forbidden()
            return
        }

        if (args.containsKey("hardly")) {
            exitProcess(0)
        }

        MyJFQL.getInstance().shutdown()
    }

    override fun complete(sender: CommandSender, line: ParsedLine): List<String>? {
        return when {
            line.wordIndex() == 1 -> listOf("hardly")
            else -> null
        }
    }

}
